src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1003,6 +1010,50 @@ public bool ProcessTransaction(Transaction transaction, int? blockHeight = null,
 
             lock (this.lockObject)
             {
+                // Do a pre-scan of the incoming transaction's inputs to see if they're used in other wallet transactions already.
+                foreach (TxIn input in transaction.Inputs)
```

Comment:
`transaction.Inputs` is iterated twice can we not use the single loop?

Reply:
I wanted to avoid making any changes to the wallet data before it's determined whether or not it's an incoming double spend transaction. Note that the remainder of the method processes outputs first, which would modify the data, before proceeding onto the inputs.

I think enumerating the inputs once with a dictionary lookup per input (on average) is not too onerous.